packages/language-chooser/src/language-chooser/index.tsx

```diff
@@ -0,0 +1,307 @@
+		( { installed } ) => ! installed
+	);
+
+	const onAddLanguage = ( locale: Language ) => {
+		setLanguages( ( current ) => [ ...current, locale ] );
+		setActiveLanguage( locale );
+	};
+
+	const isEmpty = languages.length === 0;
+	const isMoveUpDisabled =
+		! activeLanguage || languages[ 0 ]?.locale === activeLanguage?.locale;
+	const isMoveDownDisabled =
+		! activeLanguage ||
+		languages[ languages.length - 1 ]?.locale === activeLanguage?.locale;
+	const isRemoveDisabled = ! activeLanguage;
+
+	const activeLanguageIndex = languages.findIndex(
+		( { locale } ) => locale === activeLanguage?.locale
+	);
+
+	const onAdd = () => {
+		onAddLanguage( inactiveLanguage );
+
+		const installedIndex = installedLanguages.findIndex(
+			( { locale } ) => locale === inactiveLanguage.locale
+		);
+
+		const availableIndex = availableLanguages.findIndex(
+			( { locale } ) => locale === inactiveLanguage.locale
+		);
+
+		let newSelected: Language | undefined;
+
+		newSelected = installedLanguages[ installedIndex + 1 ];
+
+		if ( ! newSelected && installedLanguages[ 0 ] !== inactiveLanguage ) {
+			newSelected = installedLanguages[ 0 ];
+		}
+
+		if ( ! newSelected ) {
+			newSelected = availableLanguages[ availableIndex + 1 ];
+
+			if ( availableLanguages[ 0 ] !== inactiveLanguage ) {
+				newSelected = availableLanguages[ 0 ];
+			}
+		}
+
+		setInactiveLanguage( newSelected );
+
+		speak( __( 'Locale added to list' ) );
+	};
+
+	const onRemove = () => {
+		setActiveLanguage(
+			languages[ activeLanguageIndex + 1 ] ||
+				languages[ activeLanguageIndex - 1 ]
+		);
+
+		setLanguages( ( prevLanguages ) =>
+			prevLanguages.filter(
+				( { locale } ) => locale !== activeLanguage?.locale
+			)
+		);
+		_setLanguages( ( prevLanguages ) =>
+			prevLanguages.filter(
+				( { locale } ) => locale !== activeLanguage?.locale
+			)
+		);
+
+		speak( __( 'Locale removed from list' ) );
+
+		if ( languages.length === 1 ) {
+			let emptyMessageA11y = sprintf(
+				/* translators: %s: English (United States) */
+				__( 'No languages selected. Falling back to %s.' ),
+				'English (United States)'
+			);
+
+			if ( showOptionSiteDefault ) {
+				emptyMessageA11y = __(
+					'No languages selected. Falling back to Site Default.'
+				);
+			}
+
+			speak( emptyMessageA11y );
+		}
+	};
+
+	const onMoveUp = () => {
+		setLanguages( ( prevLanguages ) => {
+			const srcIndex = prevLanguages.findIndex(
+				( { locale } ) => locale === activeLanguage?.locale
+			);
+			return reorder(
+				Array.from( prevLanguages ),
+				srcIndex,
+				srcIndex - 1
+			);
+		} );
+
+		speak( __( 'Locale moved up' ) );
+	};
+
+	const onMoveDown = () => {
+		setLanguages( ( prevLanguages ) => {
+			const srcIndex = prevLanguages.findIndex(
+				( { locale } ) => locale === activeLanguage?.locale
+			);
+			return reorder< Language[] >(
+				Array.from( prevLanguages ),
+				srcIndex,
+				srcIndex + 1
+			);
+		} );
+
+		speak( __( 'Locale moved down' ) );
```

Comment:
Does it make sense for these "locale moved down/up" and "locale removed" strings to contain the language we're moving up or down? In my testing with assistive technology, it's not immediately clear which locale I'm moving if I forgot which one was announced as the "selected" one previously.